### PR TITLE
3.0/feature/goals n related

### PR DESCRIPTION
**Pull request by NiclasNorin**

### CODE REVIEW

File: views/partials/challenge/global-goals.blade.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<h3 class="content-width">{{$globalGoalsTitle}}</h3>
```
Comment (annalinneajohansson) [inline views/partials/challenge/global-goals.blade.php]: Is this preceded by a H2?

File: views/partials/project/contacts.blade.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<div class="o-grid-12">
+    @card
+        <div class="c-card__body">
+            <h4>{{$lang['contactsLabel']}}</h4>
```
Comment (annalinneajohansson) [inline views/partials/project/contacts.blade.php]: Is this preceded by a `<h3>`?

File: views/partials/project/files.blade.php [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<div class="o-grid-12">
+    @card
+        <div class="c-card__body u-pt-1">
+            <h4>{{$lang['filesLabel']}}</h4>
```
Comment (annalinneajohansson) [inline views/partials/project/files.blade.php]: Is this preceded by a `<h3>`?

File: views/partials/project/goals.blade.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
+@foreach ($project['impactGoals'] as $item)
+    <div class="o-grid-12">
+        @card
+            @if ($item['impact_goal'])
+                <div class="c-card__body">
+                    @if ($item['impact_goal_completed'])
+                        <h4>
```
Comment (annalinneajohansson) [inline views/partials/project/goals.blade.php]: Is this preceded by a `<h3>`?

File: views/partials/project/involvement.blade.php [hunk 5]
```diff
@@ -0,0 +1,17 @@
+@foreach ($project['residentInvolvement'] as $residentInvolement)
+    <div class="o-grid-12">
+        @card
+            <div class="c-card__body">
+                <h4>
```
Comment (annalinneajohansson) [inline views/partials/project/involvement.blade.php]: Is this preceded by a `<h3>`?

File: views/partials/project/links.blade.php [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<div class="o-grid-12">
+    @card
+        <div class="c-card__body u-pt-1">
+            <h4>{{$lang['linksLabel']}}</h4>
```
Comment (annalinneajohansson) [inline views/partials/project/links.blade.php]: Is this preceded by a `<h3>`?

File: views/partials/project/meta.blade.php [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<div class="o-grid-12">
+    @card
+        <div class="c-card__body">
+            <ul class="box-project-meta__list">
+                @foreach ($project['meta'] as $meta)
+                    <li>
+                        <h4>{{ $meta['title'] }}</h4>
```
Comment (annalinneajohansson) [inline views/partials/project/meta.blade.php]: Is this preceded by a `<h3>`?

File: views/partials/challenge/global-goals.blade.php [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+    @if (!empty($globalGoalsDescription))
+        {!!$globalGoalsDescription!!}
+    @endif
+<div class="global-goals">
+    <div class="global-goals__container">
+    @foreach($globalGoals as $item)
+        <img class="global-goals__logo" src="{{$item['featured_image']}}">
```
Comment (annalinneajohansson) [inline views/partials/challenge/global-goals.blade.php]: Image is missing alt-tag

File: views/partials/project/links.blade.php [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+            <ul>
+                @foreach ($project['links'] as $link)
+                    <li>
+                        <a target="_blank" href="{{ $link['url'] }}">{{ $link['title'] }}</a>
```
Comment (annalinneajohansson) [inline views/partials/project/links.blade.php]: Add rel="noopener" to links opened as _blank. See https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/noopener and https://sonarcloud.io/project/security_hotspots?id=helsingborg-stad_api-project-manager-integration&pullRequest=43&hotspots=AYUpvZ6Sh0kgxdFwZbka

File: source/php/Controller/Challenge.php [hunk 10]
```diff
@@ -55,6 +55,46 @@ public function singleViewController($data)
...
+            'exclude' => array(get_queried_object_id()),
+            'orderby' => 'rand'
+        ]);
+
+        if(!empty($data['relatedProjects'])) {
+            foreach($data['relatedProjects'] as $post) {
+                \ProjectManagerIntegration\Helper\AddProjectData::addPostTags($post, $post->ID);
+                \ProjectManagerIntegration\Helper\AddProjectData::addPostData($post, $post->ID);
+            }
+        };
```
Comment (annalinneajohansson) [inline source/php/Controller/Challenge.php]: Don't end if-statements with a semicolon

File: source/php/Controller/Challenge.php [hunk 11]
```diff
@@ -55,6 +55,46 @@ public function singleViewController($data)
...
+
+        if(!empty($data['relatedPosts'])) {
+            foreach($data['relatedPosts'] as $post) {
+                \ProjectManagerIntegration\Helper\AddProjectData::addPostData($post, $post->ID);
+            }
+        };
```
Comment (annalinneajohansson) [inline source/php/Controller/Challenge.php]: Don't end if-statements with a semicolon

File: views/partials/challenge/global-goals.blade.php (hunk 8 above)
Comment (NiclasNorin) [reply views/partials/challenge/global-goals.blade.php]: I will try to find an alt tag to use

**Review by nRamstedt — APPROVED** [review]

**Pull request merged**